runtime-sdk/src/context.rs

```diff
@@ -296,6 +302,9 @@ pub trait BatchContext: Context {
 
 /// Runtime SDK transaction context.
 pub trait TxContext: Context {
+    /// The index of the transaction in the block.
```

Comment:
do you mean `in the batch` instead of `in the block` here? How can client know the exact index of a transaction in the block?

Reply:
It's the batch. The client can't know the index in the batch, but it doesn't need to. The index is used to determine the nonce of the message returned by the paratime. The nonce is always included in the envelope (both ways, in fact).